### PR TITLE
Inverse kinematics

### PR DESCRIPTION
**Pull request by LemonPi**
Add inverse kinematics (IK) via basic pseudo inverse method. Comes with many options for early termination and so on.

### CODE REVIEW

File: src/pytorch_kinematics/ik.py [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    Determine the error in position and rotation between the given poses and the target poses
+
+    :param m: (N x M x 4 x 4) tensor of homogenous transforms
+    :param target_pos:
+    :param target_rot_rpy:
+    :return: (N*M, 6, 1) tensor of delta pose (dx, dy, dz, droll, dpitch, dyaw)
+    """
+    pos_diff = target_pos.unsqueeze(1) - m[:, :, :3, 3]
+    pos_diff = pos_diff.view(-1, 3, 1)
+    rot_diff = target_rot_rpy.unsqueeze(1) - rotation_conversions.matrix_to_euler_angles(m[:, :, :3, :3],
```
Comment (PeterMitrano) [inline src/pytorch_kinematics/ik.py:242]: Are we sure this is a good rotation distance? Just substracting euler angles isn't generally a good idea -- you have potential wrap-around/discontinuity issues

File: tests/test_inverse_kinematics.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    # robot frame
+    pos = torch.tensor([0.0, 0.0, 0.0], device=device)
+    rot = torch.tensor([0.0, 0.0, 0.0], device=device)
+    rob_tf = pk.Transform3d(pos=pos, rot=rot, device=device)
+
+    # world frame goal
+    M = 1000
+    # generate random goal positions
+    goal_pos = torch.rand(M, 3, device=device) * 0.5
```
[the remark this answers is not in the document]
Comment (LemonPi) [reply tests/test_inverse_kinematics.py]: Changed

File: src/pytorch_kinematics/ik.py (hunk 1 above)
Comment (LemonPi) [reply src/pytorch_kinematics/ik.py:242]: Thanks for finding this! I replaced it with unit quaternion error (the quaternion required to get from current to desired orientation), converted to an axis angle representation. 

Note that this is for computing error in end effector position and orientation to be multiplied by the jacobian pseudoinverse, so it needs to be 6D. [\dot{x} \omega] = J \dot{q} where we are approximating it to [\Delta{x} \Delta{r}] = J \Delta{q}

It's getting much better convergence now.

**Comment by LemonPi** [comment]
mujoco_menagerie seems to need jax (workflow file)
```
      - name: Clone mujoco_menagerie repository into the tests/ folder
        run: |
          git clone https://github.com/google-deepmind/mujoco_menagerie
        working-directory: ${{ runner.workspace }}/pytorch_kinematics/tests
```
how do you think we should include it? Something like:
```
      - name: Clone mujoco_menagerie repository into the tests/ folder
        run: |
          python -m pip install "jax[cpu]"
          git clone https://github.com/google-deepmind/mujoco_menagerie
        working-directory: ${{ runner.workspace }}/pytorch_kinematics/tests
```

File: pyproject.toml [hunk 3]
```diff
@@ -46,10 +46,16 @@ dependencies = [
...
+    'matplotlib',
+    'pytorch_seed',
 ]
 
 [project.optional-dependencies]
-test = ["pytest"]
+test = [
+    "pytest",
+    "pybullet",
+    "jax",
```
Comment (PeterMitrano) [inline pyproject.toml:57]: do we actually need this? I don't see it being used

**Review by PeterMitrano — APPROVED** [review]
just one small comment, otherwise LGTM

File: pyproject.toml (hunk 3 above)
Comment (LemonPi) [reply pyproject.toml]: This has been removed a couple commits back (only needed for mujoco menagerie)
